app/controllers/items_controller.rb

```diff
@@ -63,7 +61,17 @@ def show
 
   def update
     @item = current_organization.items.find(params[:id])
-    if @item.update(item_params)
+    @item.attributes = item_params
+
+    deactivated = @item.active_changed? && !@item.active
+    if deactivated && !@item.can_deactivate?
+      @base_items = BaseItem.without_kit.alphabetized
+      flash[:error] = "Can't deactivate this item - it is currently assigned to either an active kit or a storage location!"
+      render action: :edit
+      return
+    end
+
+    if @item.save
```

Comment:
Why did  we change from update to save?

Reply:
`update` both sets the parameters and saves in a single function. In this case we're setting the parameters early (to check if the `active` parameter changed) and then saving it after our check.